lib/utils/location_permission_utils.dart

```diff
@@ -40,9 +40,11 @@ Future<void> openAppSettings() async {
   await Geolocator.openAppSettings();
 }
 
+// TODO: use GlobalKey to access the context
```

Comment:
This should be removed? because u already use globalkey.

Also regarding this new method to fix using context inside async. We have this method "isMounted", that was a stupid fix for the issue, I think we should refactor this and remove all instances of isMounted and replace them with global keys.

Reply:
Yeah, thats what the TODO

Reply:
oh ok